### ISSUE TITLE
Can't get env flag to target usgovernment

### ISSUE THREAD

**Issue by jasperstone** [Azure/aztfexport#392]
Can you provide an example of the env flag? I don't see one in the help output and I can't get aztfexport to target a cloud other than public.

With this command:
```
aztfexport.exe rg --env usgovernment <rgname>
```
I get the following...
```
Incorrect Usage: flag provided but not defined: -env
```

I have tried the following variations:
```
--env usgovernment
--env=usgovernment
-env usgovernment
env usgovernment
```

When setting this environment variable:
```
$AZTFEXPORT_ENV=usgovernment
```
And executing this command:
```
aztfexport.exe rg <rgname>
```
I get the following (url makes me think it's targeting public)...

```
listing resource set: executing ARG query "Resources | where resourceGroup =~ \"<rgname>" | order by id desc": POST
  https://management.azure.com/providers/Microsoft.ResourceGraph/resources
  --------------------------------------------------------------------------------
  RESPONSE 400: 400 Bad Request
  ERROR CODE: BadRequest
  --------------------------------------------------------------------------------
  {
    "error": {
      "code": "BadRequest",
      "message": "Please provide below info when asking for support: timestamp = 2023-04-20T14:42:08.5137060Z, correlationId =
  <redacted>",
      "details": [
        {
          "code": "NoValidSubscriptionsInQueryRequest",
          "message": "There must be at least one subscription that is eligible to contain resources. Given: <valid usgov subscription id>."
        }
      ]
    }
  }
  --------------------------------------------------------------------------------
```

**Comment by magodo** [comment]
@jasperstone Which `aztfexport` version you were using? The `--env` option is not yet released in v0.11.0, while you can install it from source by: `go install github.com/Azure/aztfexport@main`.

**Comment by jasperstone** [comment]
I am using v0.11.0. The readme seems to indicate that the --env option is available now.
https://github.com/Azure/aztfexport#cloud-environment

Setting the ARM_ENVIRONMENT variable works if I export a resource. Win!

However, if I try to export a resource group, the subscription id is always blank, even if I use the "-s <subscription_id>" parameter or set ARM_SUBSCRIPTION_ID. I get a similar error to the one posted above (400 Bad Request: NoValidSubscriptionsInQueryRequest). The url looks good (management.usgovcloudapi.net) but at the end of the error message is says: Given: '' <- empty subscription id.

**Comment by magodo** [comment]
@jasperstone The presence of the section "cloud-environment" in the README indicates it is available in `main` branch, not necessarily mean it is available in the latest release.

Regarding your observation, I'm not sure what goes wrong. Could you check the ARG request in the log by specifying `--log-level TRACE --log-path <log path>` options?

**Comment by Flightdeck73** [comment]
using aztfexport version v0.12.0(fc6e1ff)

 `aztfexport rg --non-interactive --env usgovernment dev-resource-group`

when executing this command i get an empty provider.tf, terraform.tf and empty state file, are my expectations wrong or is it supposed to` tf import then add`?

The **resourceMapping.json** has most of the resources _( doesn't like vm/disks )_ 

**Error: retrieving Virtual Machine** "ima-win-vm" (Resource Group "dev-resource-group"): compute.VirtualMachinesClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="SubscriptionNotFound" Message="The subscription 'xxx' could not be found."

While attempting to import an existing object to
"**azurerm_managed_disk.res-4**", the provider detected that no object exists
with the given id. Only pre-existing objects can be imported; check that the
id is correct and that it is associated with the provider's configured region
or endpoint, or use "terraform apply" to create a new remote object for this
resource.

**Comment by magodo** [comment]
Fixed by #546
